controllers/search_controller.rb

```diff
@@ -13,7 +13,70 @@ class SearchController < ApplicationController
     end
 
     get '/metadata' do
-      process_search
+      if params[:query].nil? && params[:q].nil?
+        raise error 400, "The search query must be provided via /search?q=<query>[&page=<pagenum>&pagesize=<pagesize>] /search?query=<query>[&page=<pagenum>&pagesize=<pagesize>]"
+      end
+
+      query = params[:query] || params[:q]
+      page, page_size = page_params
+
+      # Filters
+      acronyms = params.fetch("acronyms", "").split(',')
+      status = params.fetch("status", "").split(',')
+
+
+      # Query Fields (Boosting)
+      qf = [
+        "acronym_text^50",
+        "description_text^40",
+        "name_text^30",
+        "homepage_t^10",
+        "version_t^5"
+      ].join(" ")
+
+
+      fq = []
+      fq << acronyms.map { |x| "acronym_text:\"#{x}\"" }.join(" OR ") unless acronyms.empty?
+      fq << status.map { |x| "status_t:#{x}" }.join(' OR ') unless status.blank?
+      
+      resp = search(LinkedData::Models::Ontology,
+                    query,
+                    fq: fq, qf: qf,
+                    page: page, page_size: page_size,
+                    sort: "score desc, acronym_sort asc, name_sort asc")
+
+      result = {}
+      total_found = resp.length
+
+      # Build a hash of ontology artefacts by merging information from two resource types:
+      # "ontology" provides basic metadata (acronym, name), while "ontology_submission" adds details (description, homepage, version)
+      # Entries are merged by resource ID to avoid duplication.
+      resp.each do |doc|
+        doc = doc.symbolize_keys
+        resource_model = doc[:resource_model]
+        case resource_model
+        when "ontology"
+          artefact_id = doc[:resource_id]
+          result[artefact_id] ||= {
+            id: "#{Goo.id_prefix}artefacts/#{doc[:acronym_text]}",
+            acronym: doc[:acronym_text]
+          }
+          result[artefact_id][:name] ||= doc[:name_text]
+        when "ontology_submission"
```

Comment:
filter the search metadata to search only `ontology_submission`, because it does include already `ontology` metadata.